subsys/pm/Kconfig

```diff
@@ -82,6 +82,14 @@ config PM_DEVICE_POWER_DOMAIN
 	  devices that depend on a domain will be notified when this
 	  domain is suspended or resumed.
 
+config PM_DEVICE_POWER_DOMAIN_DYNAMIC
```

Comment:
Make the build system updates less specific to this particular feature. Add a (hidden?) option (`config DEVICE_DYNAMIC_HANDLES` or similar) which this symbol can select and can drive the changes in `device.h` and the linker scripts.

Reply:
That was something that I initially had thought and decided to take the current path to avoid and additional symbol, but that is something I definitely can change.